bftengine/src/bcstatetransfer/SourceSelector.cpp

```diff
@@ -34,12 +34,22 @@ void SourceSelector::setFetchingTimeStamp(uint64_t currTimeMilli, bool retransmi
 void SourceSelector::removeCurrentReplica() {
   preferredReplicas_.erase(currentReplica_);
   currentReplica_ = NO_REPLICA;
+  checkAndRefillPreferredReplicas();
   receivedValidBlockFromSrc_ = false;
   setSourceSelectionTime(0);
   metrics_.current_source_replica_.Get().Set(currentReplica_);
   metrics_.preferred_replicas_.Get().Set(preferredReplicasToString());
 }
 
+void SourceSelector::removePreferredReplica(uint16_t replicaId) {
+  if (preferredReplicas_.find(replicaId) == preferredReplicas_.end()) {
+    LOG_WARN(logger_, "Not found" << KVLOG(replicaId));
```

Comment:
nit: rethink again if you want to just return, or since replicaId is not in preferredReplicas_ you should terminate?

Reply:
I think can be ignored for now as anyways SS needs to be refactored.